badger/cmd/info.go

```diff
@@ -59,39 +63,78 @@ info. It also prints info about missing/extra files, and general information abo
 files (which are not referenced by the manifest).  Use this tool to report any issues about Badger
 to the Dgraph team.
 `,
-	Run: func(cmd *cobra.Command, args []string) {
-		err := printInfo(sstDir, vlogDir)
-		if err != nil {
-			fmt.Println("Error:", err.Error())
-			os.Exit(1)
-		}
-		if !opt.showTables {
-			return
-		}
-		// Open DB
-		opts := badger.DefaultOptions
-		opts.TableLoadingMode = options.MemoryMap
-		opts.Dir = sstDir
-		opts.ValueDir = vlogDir
-		opts.ReadOnly = true
-
-		db, err := badger.Open(opts)
-		if err != nil {
-			fmt.Println("Error:", err.Error())
-			os.Exit(1)
-		}
-		defer db.Close()
+	RunE: handleInfo,
+}
+
+func handleInfo(cmd *cobra.Command, args []string) error {
+	err := printInfo(sstDir, vlogDir)
+	if err != nil {
+		fmt.Println("Error:", err.Error())
+		os.Exit(1)
+	}
+
+	// Open DB
+	opts := badger.DefaultOptions
+	opts.TableLoadingMode = options.MemoryMap
+	opts.Dir = sstDir
+	opts.ValueDir = vlogDir
+	opts.ReadOnly = true
+
+	db, err := badger.Open(opts)
+	if err != nil {
+		fmt.Println("Error:", err.Error())
+		os.Exit(1)
+	}
+	defer db.Close()
 
+	if opt.showTables {
 		err = tableInfo(sstDir, vlogDir, db)
 		if err != nil {
 			fmt.Println("Error:", err.Error())
 			os.Exit(1)
 		}
-		if opt.sizeHistogram {
-			// use prefix as nil since we want to list all keys
-			db.PrintHistogram(nil)
+	}
+
+	if opt.sizeHistogram {
+		db.PrintHistogram([]byte(opt.withPrefix))
+	}
+
+	if opt.dumpData {
+		if err := dumpData(db, opt.withPrefix); err != nil {
+			return err
```

Comment:
The bahaviour here is different than what we do with error in rest of the places in this function.

Reply:
Fixed.